esmvalcore/_recipe.py

```diff
@@ -694,6 +703,70 @@ def _update_extract_shape(settings, config_user):
         check.extract_shape(settings['extract_shape'])
 
 
+def _parse_period(timerange):
+    start_year = None
+    end_year = None
+    if timerange.split('/')[0].startswith('P'):
+        try:
+            end_year = isodate.parse_datetime(timerange.split('/')[1]).year
+        except isodate.ISO8601Error:
+            end_year = isodate.parse_date(timerange.split('/')[1]).year
+        delta = int(isodate.parse_duration(timerange.split('/')[0]).years)
+        start_year = end_year - delta
+    elif timerange.split('/')[1].startswith('P'):
+        try:
+            start_year = isodate.parse_datetime(timerange.split('/')[0]).year
+        except isodate.ISO8601Error:
+            start_year = isodate.parse_date(timerange.split('/')[0]).year
+        delta = int(isodate.parse_duration(timerange.split('/')[1]).years)
+        end_year = start_year + delta
+    return start_year, end_year
+
+
+def _update_timerange(variable, settings, config_user):
+    if 'timerange' not in variable:
+        return
+
+    timerange = variable.get('timerange')
+    check.valid_time_selection(timerange)
+
+    step = 'clip_start_end_year'
+    if '*' in timerange:
+        (files, _, _) = _find_input_files(variable, config_user['rootpath'],
+                                          config_user['drs'])
+        intervals = [get_start_end_year(name) for name in files]
+
+        min_date = min(intervals)[2]
+        max_date = max(intervals)[3]
+
+        if timerange == '*':
+            timerange = f'{min_date}/{max_date}'
+        if '*' in timerange.split('/')[0]:
+            timerange = timerange.replace('*', min_date)
+        if '*' in timerange.split('/')[1]:
+            timerange = timerange.replace('*', max_date)
+
+    start_year, end_year = _parse_period(timerange)
+
+    if start_year is None and end_year is None:
+        start_year = int(timerange.split('/')[0][0:4])
+        end_year = int(timerange.split('/')[1][0:4])
+
+    variable.update({
+        'timerange': timerange,
+        'start_year': start_year,
+        'end_year': end_year
+    })
+
+    settings[step]['timerange'] = timerange
+    settings[step]['start_year'] = start_year
+    settings[step]['end_year'] = end_year
+
+    timerange = timerange.replace('/', '-')
+    filename = variable['filename'].replace('.nc', f'_{timerange}.nc')
```

Comment:
This looks dodgy, I think you would need to make sure that the right value is passed into the function `get_output_file` from _data_finder.py in the first place, instead of changing it later on